core/src/main/kotlin/cash/grammar/kotlindsl/model/gradle/DependencyContainer.kt

```diff
@@ -18,26 +19,15 @@ public class DependencyContainer(
   public val elements: List<Any>,
 ) {
 
-  public fun getDependencyDeclarations(): List<DependencyDeclaration> {
-    return elements.filterIsInstance<DependencyDeclaration>()
-  }
-
-  @Deprecated(
-    message = "use getExpressions",
-    replaceWith = ReplaceWith("getExpressions()")
-  )
-  public fun getNonDeclarations(): List<String> {
-    return getExpressions()
+  public fun getDependencyDeclarationsWithContext(): List<DependencyDeclarationWithContext> {
+    return elements.filterIsInstance<DependencyDeclarationWithContext>()
   }
 
-  /**
-   * A common example of an expression in a dependencies block is
-   * ```
-   * add("extraImplementation", "com.foo:bar:1.0")
-   * ```
-   */
-  public fun getExpressions(): List<String> {
-    return elements.filterIsInstance<String>()
+  public fun getDependencyDeclarations(): List<DependencyDeclaration> {
+    val declarationsWithContext = getDependencyDeclarationsWithContext().map { it.declaration }
+    return declarationsWithContext.ifEmpty {
+      elements.filterIsInstance<DependencyDeclaration>()
+    }
```

Comment:
Are there cases where "raw" `DependencyDeclarations` are in the list of `elements`? This seems unnecessary.

Reply:
After introducing `DependencyElement` module, we no longer need to do this. I removed it